code/datums/ai/dog/dog_controller.dm

```diff
@@ -131,7 +131,7 @@
 	var/obj/item/thrown_thing = carbon_thrower.get_active_held_item()
 	if(!isitem(thrown_thing))
 		return
-	if(blackboard[BB_FETCH_IGNORE_LIST][thrown_thing])
+	if(blackboard[BB_FETCH_IGNORE_LIST][WEAKREF(thrown_thing)])
```

Comment:
It doesn't look like you call `resolve()` on this anywhere. If I am correct, you should just use `REF`, which will just give you a static ID.

The downside is when the thing gets deleted, its memory will be overridden and the ref could refer to something else (except for mobs), but it doesn't look like it should be a problem here.

Reply:
See discussion about scanned, tho yeah, ref might just be better here